costmanagementserv/src/main/java/com/costs/costmanagement/CostmanagementservApplication.java

```diff
@@ -0,0 +1,130 @@
+import org.springframework.boot.SpringApplication;
+import org.springframework.boot.autoconfigure.SpringBootApplication;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+import org.springframework.web.servlet.support.ServletUriComponentsBuilder;
+
+import javax.validation.Valid;
+import java.net.URI;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+@SpringBootApplication
+@RestController
+public class CostmanagementservApplication {
+	@Autowired
+	private CostsDbRepository costsDbRepository;
+
+	public static void main(String[] args) {
+		SpringApplication.run(CostmanagementservApplication.class, args);
+	}
+	/**
+	 *
+	 * @param id show id
+	 * @return list of aggregated episode cost, excluding amortized costs.
+	 *
+	 * This should work for both TASK 1 & 2
+	 */
+
+	@GetMapping("/costs/{id}")
+	public ResponseEntity<List<ShowEpisodeCostAPIModel>> getBasicCost(@PathVariable Long id) {
+		List<ShowEpisodeCost> doObjects = this.costsDbRepository.getAggregatEpisodeCostsForShowWithoutAmortizedCost(id);
+		List<ShowEpisodeCostAPIModel> costReport = doObjects.stream().map(doObject ->{
+			ShowEpisodeCostAPIModel apiModel = new ShowEpisodeCostAPIModel();
+			apiModel.setAmount(doObject.getAmount().toString());
+			apiModel.setEpisode_code(doObject.getEpisodeCd().toString());
+			apiModel.setId(doObject.getId().toString());
+					return apiModel; }).collect(Collectors.toList());
+
+		if (costReport.size() == 0) {
+			return ResponseEntity.notFound().build();
+		} else {
+			return ResponseEntity.ok(costReport);
+		}
+	}
+	/**
+	 *
+	 * @param showEpisodeCostListStr json object include list of show id, episode code, and cost amount
+	 * @return created episode cost object
+	 *
+	 *  This should work for both TASK 1 & 2
+	 */
+	@PostMapping("/costs")
+	@ResponseBody
+	public ResponseEntity<List<ShowEpisodeCostAPIModel>> createCost(@RequestBody @Valid String showEpisodeCostListStr) {
+		ObjectMapper objectMapper = new ObjectMapper();
+		List<ShowEpisodeCostAPIModel> showEpisodeCostList;
+		try {
+			showEpisodeCostList = objectMapper.readValue(showEpisodeCostListStr, new TypeReference<List<ShowEpisodeCostAPIModel>>() {
+			});
+		} catch (JsonProcessingException e){
+			return ResponseEntity.badRequest().build();
+		}
+		List<ShowEpisodeCostAPIModel> createdList = new ArrayList<>();
+		for(ShowEpisodeCostAPIModel showEpisodeCostAPIModel:showEpisodeCostList) {
+			if(showEpisodeCostAPIModel.getEpisode_code() == null || showEpisodeCostAPIModel.getAmount() == null
+					|| showEpisodeCostAPIModel.getId() == null || showEpisodeCostAPIModel.getEpisode_code().length() != 3){
+				return ResponseEntity.badRequest().build();
+			}
+			Optional<ShowEpisodeCost> createdEpisodeCost = this.costsDbRepository.createCost(Long.parseLong(showEpisodeCostAPIModel.getId()), showEpisodeCostAPIModel.getEpisode_code(),
```

Comment:
we can use batch insertion on the db side to reduce db connection costs. depend on how many records we will need to insert in each request, if say we have to insert 1000 in a request, we can break it down into smaller chunks before doing batch insertion. this number can vary. let's say we do 100 in one batch insertion, then we call batchInsert 10 times for this 1000 records. reason to break it down is because 100 per insertion is transactional, meaning one fail, all fail. this is to reduce chances of failing. also you might have db restrictions on how many rows you can commit each time.

Reply:
I updated the create to use batch insert. Assumption here is the list I get is < 100. If we have thousands, we will need to break it down to smaller chunks.